### DISCUSSION TITLE
miglior modo per utilizzare zigbee/matter

### DISCUSSION THREAD

**Discussion by barbatrukko** [indomus/forum#2612] [HUB: Home Assistant]
Salve,
al momento il mio Home Assistant è installato su un RasPi3B con HassOS.
Ho visto che ci sono diversi dongle, pennette, etc per potersi affacciare al mondo Zigbee/matter. 
Qualcuna è compatibile con raspi3 ? 
Inoltre io spero a breve di passare ad una istanza virtualizzata su un nuc (non un nuc intel, ma semplicemente un miniserver non arm) su cui far girare un proxmox/esxi e gestire più VM in modo da installare a parte (con scrittura su SSD) infuxDB e Grafana ed altri software che al momento girano su altri raspPi4.

Non so se questi dongle zigbee/matter sono specifici per determinati hardware. Qualcuno può aiutarmi a scegliere?
Se lo prendo per raspi3 poi non lo posso usare sul nuc? o basta che ci sia uan porta USB e poi il driver c'è per tutti i S.O. ?

**Comment by indomus** [comment]
versione breve: prendi questa https://indomus.it/componenti/itead-sonoff-zigbee-3-0-usb-dongle-plus-v2-zbdongle-e/
versione lunga: leggi con attenzione questa pagina: https://indomus.it/componenti/zigbee/

se prendi un mini pc/nuc non ti serve usare alcuna VM, operi precisamente come raspberry.

**Reply by barbatrukko** [reply]
Quindi se prendo il dongle che suggerisci, posso al momento collegarlo al raspi3 e se un domani passo al nuc lo posso usare anche sul nuc?

Se passo al nuc è per far girare più vm con software diverso, per questo uso le VM, devo spostare cose che sono al momento ospitate su raspi4 diversi e che non possono coesistere.

**Reply by indomus** [reply]
se devi far girare home assistant, zigbee2mqtt, influx e grafana e usi VM, stai sbagliando approccio.
quel dongle lo usi con qualsiasi host.

**Reply by barbatrukko** [reply]
Ok, grazie per le info sul dongle.
In merito alle VM, su una girerà HASSOS, su una un ubuntu su cui metterò a parte grafana e influxdb, e su altre VM altri software che servono a me in altri contesti.
Capisco che su un hardware carrozzato tutto lo stack ha potrebbe essere dockerizzato sullo stesso server, ma a me serve far girare software diversi che non posso tenere insieme per altri motivi.

**Reply by indomus** [reply]
ah ok, spiegata così è un po' diversa :) ok ok

**Reply by barbatrukko** [reply]
Domanda più complicata: attualmente il mio HA gestisce via rete wired/wireless sia i dispostivi di casa mia, sia alcuni dispositivi che stanno 2 piani sotto di me in un altro appartamento connesso alla mia stessa rete (sia wireless che wired: fra il mio appartamento e l'altro c'è un piano in mezzo). Volendo utilizzare anche in quell'appartamento dispositivi Zigbee esiste un dispositivo da connettere alla rete wired o wireless del secondo appartamento che mi espanda fino a li il segnale zigbee? Non sono sicuro che ci arrivi direttamente il mio dal mio appartamento.... So che i device zigbee si fanno da espansione a vicenda... ma io ne userei alcuni al piano 2 ed alcuni al piano terra, in mezzo non c'è nulla...

**Comment by barbatrukko** [comment]
Salve,
ho acquistato il dongle di cui sopra e l'ho collegato (lo vedo come ttyACM0.

La mia domanda è: poso usarlo così o devo per forza riprogrammare il firmware come indicato qui?

https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zbdongle-e-zigbee-3-0-usb-dongle-plus-v2/

**Reply by KC-inDomus** [reply]
puoi usarlo così com'è

**Comment by barbatrukko** [comment]
Attualmente ho questo errore all'avvio di Zigbee2mqtt
Stiamo parlando di RasPi3 e HassOS

```
[core-ssh 2023-09-07.21-56-48]$ cat log.txt
info  2023-09-07 21:56:48: Logging to console and directory: '/config/zigbee2mqtt/log/2023-09-07.21-56-48' filename: log.txt
info  2023-09-07 21:56:48: Starting Zigbee2MQTT version 1.33.0 (commit #unknown)
info  2023-09-07 21:56:48: Starting zigbee-herdsman (0.18.5)
error 2023-09-07 21:56:49: Error while starting zigbee-herdsman
error 2023-09-07 21:56:49: Failed to start zigbee
error 2023-09-07 21:56:49: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2023-09-07 21:56:49: Exiting...
error 2023-09-07 21:56:49: Error: spawn udevadm ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:285:19)
    at onErrorNT (node:internal/child_process:485:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

```

**Reply by KC-inDomus** [reply]
hai sbagliato la configurazione di zigbee2mqtt

**Reply by barbatrukko** [reply]
Ho usato quella della guida indicata.
Comunque poi sono riuscito a far partire il tutto prendendo la configurazione online dal sito di zigbeemqtt e facendo un paio di modifiche (tipo NON indicando la parte mqtt perché sul sito diceva di non usare questa sezione se si usa un mosquitto interno).

la mia domanda ora  è: questa guida uscita oggi 
https://indomus.it/guide/come-installare-e-configurare-silicon-labs-multiprotocol-per-gestire-coordinator-multipan-rcp/

è una alternativa a quella seguita in questa discussione?

che differenza c'è fra zigbee2mqtt e MultiPAN RCP?

**Reply by KC-inDomus** [reply]
se fai questa domanda significa che hai concetti molto confusi in testa. leggi bene l'Abstract della guida uscita oggi, dovrebbe chiarirtisi.

ANZI: leggiti [anche questa](https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zbdongle-e-zigbee-3-0-usb-dongle-plus-v2/)

**Reply by barbatrukko** [reply]
ok, quindi se ho ben capito, cambiando il firmware del dongle e installando il modulo software intermedio , faccio puntare mqtt al ip/porta del software intermedio e non direttamente alla pennetta.
ok, ci provo

**Reply by Calabash78** [reply]
ciao sei riuscito a far dialogare la chiavetta dongle_E con firmware rcp con il demone silicon labs e zigbeetomqqt?
hai modificato stringhe su configuration.yaml?

**Comment by barbatrukko** [comment]
Ancora no,
mi sono prefissato di smanettarci questo fine settimana in quanto il
precedente non ho potuto.
Appena faccio le prove comunque posto qui i risultati.

Il giorno mer 20 set 2023 alle ore 18:19 Calabash78 <
***@***.***> ha scritto:
